### PR TITLE
ccfri-4027 - fix document upload bugs and code refactor

### PR DESCRIPTION
**Pull request by vietle-cgi**
# Description

## Checklist

- [ ] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [ ] I have checked that unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: frontend/src/components/SupportingDocumentUpload.vue [hunk 1]
```diff
@@ -308,51 +296,45 @@ export default {
...
       'createChangeAction',
       'deleteChangeAction',
       'addChangeNotificationId',
       'deleteChangeNotificationId',
     ]),
     ...mapActions(useNavBarStore, ['forceNavBarRefresh']),
+    loadSupportingDocuments() {
+      this.uploadedSupportingDocuments = this.applicationUploadedDocuments?.filter((document) => {
+        return (
+          this.navBarList?.findIndex((item) => item.facilityId === document.facilityId) > -1 &&
```
Comment (weskubo-cgi) [inline frontend/src/components/SupportingDocumentUpload.vue:333]: Would some() be cleaner here?

File: frontend/src/components/SupportingDocumentUpload.vue [hunk 2]
```diff
@@ -419,17 +401,17 @@ export default {
       this.isProcessing = true;
       try {
         await this.processDocumentFileDelete();
-        const newFilesAdded = this.uploadedSupportingDocuments.filter((el) => !!el.id);
-        if (newFilesAdded.length > 0) {
+        const newFilesAdded = this.uploadedSupportingDocuments?.filter((el) => !!el.id);
+        if (newFilesAdded?.length > 0) {
```
Comment (weskubo-cgi) [inline frontend/src/components/SupportingDocumentUpload.vue:405]: Could switch to isEmpty(). But the code is already a mix so fine as is as well.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: frontend/src/components/SupportingDocumentUpload.vue (hunk 1 above)
Comment (vietle-cgi) [reply frontend/src/components/SupportingDocumentUpload.vue]: agreed. it's updated.

File: frontend/src/components/SupportingDocumentUpload.vue (hunk 2 above)
Comment (vietle-cgi) [reply frontend/src/components/SupportingDocumentUpload.vue]: I've updated all .length > 0 to use isEmpty in this file

File: frontend/src/components/RFI/RFIDocumentUpload.vue [hunk 3]
```diff
@@ -198,31 +181,12 @@ export default {
       },
     },
   },
+  created() {
+    this.FILE_REQUIREMENTS_TEXT = FILE_REQUIREMENTS_TEXT;
+    this.FILE_TYPES_ACCEPT = FILE_TYPES_ACCEPT;
```
[the remark this answers is not in the document]
Comment (weskubo-cgi) [reply frontend/src/components/RFI/RFIDocumentUpload.vue:186]: Make sense. Thanks.

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**